src/server/models/mediaquery.js

```diff
@@ -3,17 +3,78 @@ const client = require('./db');
 
 const { DB_DATABASE } = process.env;
 
-module.exports.getCastsByCourse = async courseLabel => {
-  const castCollection = client.db(DB_DATABASE).collection('bruincastmedia');
-  const toBeReturned = await castCollection
-    .find({ classShortname: courseLabel })
+module.exports.getCastsByCourse = async (dbCollection, courseLabel) => {
+  /*
+   * Aggregation Steps:
+   * 1. Match by courseLabel
+   * 2. Sort course records by date
+   * 3. Group buckets of records by 'week' field
+   *
+   * Due to the way MongoDB aggregation works in how it buckets these items,
+   * there needs to be another boundary after 'Finals' in order for 'Finals'
+   * items to be grouped together
+   */
+  const aggregation = [
+    {
+      $match: {
+        classShortname: courseLabel,
+      },
+    },
+    {
+      $sort: {
+        date: 1,
+      },
+    },
+    {
+      $bucket: {
+        groupBy: '$week',
+        boundaries: [
+          '0',
+          '1',
+          '10',
```

Comment:
@chern Oh strange.. Why is 10 in this order?

Reply:
The string "10" comes before "2"… putting "10" at the end will cause Mongo to throw an error that the boundaries should be ascending order

Reply:
@chern Can the week then be stored a an INT? We can map 88 to Finals week, like the Registrar has it.

Reply:
@chern But if 10 is stored this way.. when you query for courses.. wouldn't it display the weeks in this order?

Reply:
@rlorenzo I had previously rearranged Week 10's placement in the array manually in code after getting the response, but storing week as int will indeed make everything simpler